activation/nipost_test.go

```diff
@@ -807,6 +807,299 @@ func TestNIPSTBuilder_PoetUnstable(t *testing.T) {
 	})
 }
 
+// TestNIPoSTBuilder_PoETConfigChange checks if
+// it properly detects added/deleted PoET services and re-registers if needed.
+func TestNIPoSTBuilder_PoETConfigChange(t *testing.T) {
+	t.Parallel()
+
+	sig, err := signing.NewEdSigner()
+	require.NoError(t, err)
+
+	challenge := types.NIPostChallenge{
+		PublishEpoch: postGenesisEpoch + 2,
+	}
+
+	challengeHash := wire.NIPostChallengeToWireV1(&challenge).Hash()
+
+	const (
+		poetProverAddr  = "http://localhost:9999"
+		poetProverAddr2 = "http://localhost:9988"
+	)
+
+	t.Run("1 poet deleted BEFORE round started -> continue with submitted registration", func(t *testing.T) {
+		db := localsql.InMemory()
+		ctrl := gomock.NewController(t)
+
+		poet := NewMockPoetService(ctrl)
+		poet.EXPECT().
+			Submit(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).
+			AnyTimes().
+			Return(&types.PoetRound{}, nil)
+		poet.EXPECT().Address().AnyTimes().Return(poetProverAddr).AnyTimes()
```

Comment:
NIT: `AnyTimes()` only needs to be specified once.

```suggestion
		poet.EXPECT().Address().Return(poetProverAddr).AnyTimes()
```